baselines/feddebug/README.md

```diff
@@ -0,0 +1,221 @@
+Check the documentation for the different ways of installing `pyenv`, but one easy way is using the [automatic installer](https://github.com/pyenv/pyenv-installer):
+
+```bash
+curl https://pyenv.run | bash # then, don't forget links to your .bashrc/.zshrc
+```
+
+You can then install any Python version with `pyenv install 3.10.14` Then, in order to use FedDebug baseline, you'd do the following:
+
+```bash
+# cd to your feddebug directory (i.e. where the `pyproject.toml` is)
+pyenv local 3.10.14
+poetry env use 3.10.14 # set that version for poetry
+
+# run this from the same directory as the `pyproject.toml` file is
+poetry install
+poetry shell
+
+# check the python version by running the following command
+python --version # it should be >=3.10.14
+```
+
+This will create a basic Python environment with just Flower and additional packages, including those needed for simulation. Now you are inside your environment (pretty much as when you use `virtualenv` or `conda`).
+
+## Running the Experiments
+
+> [!NOTE]
+> You can run almost any evaluation from the paper by changing the parameters in `conf/base.yaml`. Also, you can change the resources (per client CPU and GPU) in conf/base.yaml to speed up the simulation. Please check the Flower simulation guide for more details ([Flower Framework main](https://flower.ai/docs/framework/how-to-run-simulations.html)).
+
+The following command will run the default experimental setting in `conf/base.yaml` (LeNet, MNIST, with a total of 10 clients, where client-0 is malicious). FedDebug will identify client-0 as the malicious client. **The experiment took on average 60 seconds to complete.**
+
+```bash
+python -m feddebug.main device=cpu
+```  
+
+Output of the last round will show the FedDebug output as follows:
+
+```log
+... 
+[2024-10-24 12:25:48,758][flwr][INFO] - ***FedDebug Output Round 5 ***
+[2024-10-24 12:25:48,758][flwr][INFO] - True Malicious Clients (Ground Truth) = ['0']
+[2024-10-24 12:25:48,758][flwr][INFO] - Total Random Inputs = 10
+[2024-10-24 12:25:48,758][flwr][INFO] - Predicted Malicious Clients = {'0': 1.0}
+[2024-10-24 12:25:48,758][flwr][INFO] - FedDebug Localization Accuracy = 100.0
+[2024-10-24 12:25:49,577][flwr][INFO] - fit progress: (5, 0.00015518503449857236, {'accuracy': 0.978, 'loss': 0.00015518503449857236, 'round': 5}, 39.02993568999227)
+[2024-10-24 12:25:49,577][flwr][INFO] - configure_evaluate: no clients selected, skipping evaluation
+[2024-10-24 12:25:49,577][flwr][INFO] - 
+[2024-10-24 12:25:49,577][flwr][INFO] - [SUMMARY]
+...
+
+```
+It predicts the malicious client(s) with 100% accuracy. `Predicted Malicious Clients = {'0': 1.0}`  means that client-0 is predicted as the malicious client with 1.0 probability. It will also generate a graph `iid-lenet-mnist.png` as shown below: 
+![FedDebug Malicious Client Localization IID-LeNet-MNIST](figures/iid-lenet-mnist.png)
+
+### a. Multiple Malicious Clients
```

Comment:
I feel there is a need for a mini-paragraph before the rest of the README.md so to briefly introduce that you are going to show 3 scenarios. Could we also not use `a.b.c.` and instead use `1.2.3`? Could you point each of them to some parts of your paper?

Reply:
I have applied the feedback and added a short paragraph. Thank you